test/coupler_compatibility.jl

```diff
@@ -0,0 +1,171 @@
+using Test
+import Thermodynamics as TD
+import SurfaceFluxes as SF
+import StaticArrays as SA
+import ClimaAtmos as CA
+import ClimaAtmos.Parameters as CAP
+import ClimaAtmos.SurfaceConditions: projected_vector_data, CT1, CT2
+import ClimaCore: Spaces, Fields
+import ClimaCore.Utilities: half
+
+const z0m = 1e-3
+const z0b = 1e-5
+const gustiness = 1
+const beta = 1
+const T1 = 300
+const T2 = 290
+
+# This file contains two tests that illustrate how the coupler can modify the
+# surface temperature in ClimaAtmos.
+
+# In the first test, the ClimaAtmos "cache" p is overwritten so that it contains
+# a surface field specified by the coupler, and then the internal function
+# set_precomputed_quantities! is called to verify that this surface field is
+# used correctly.
+
+# In the second test, the cache is not overwritten. Instead, the coupler defines
+# its own surface fields, and then it copies the data from those fields into the
+# cache by calling set_surface_conditions!.
```

Comment:
Could you add an inline comment showing where the coupler defines the surface fields?

Reply:
I feel like that will go stale very quickly.. Would it be helpful to just provide a link to the coupler repo?

Reply:
I guess I'm just confused about how we can be using a function from the coupler here since we aren't dependent on it. When we say "the coupler defines its own surface fields", are we using a function defined here that simulates the coupler's behavior (maybe `update_surface_fields!`)?

Reply:
I think Julia has a good point. We should clarify

Reply:
Maybe @dennisYatunin can clarify?

Reply:
Oh, sorry if this was unclear. I meant that, if the coupler were to use the design pattern shown in this section, it would need to define its own surface fields, and then it would need to copy the data from those fields into the cache. I do not believe the coupler is currently doing this; instead, it is using the design pattern shown in the preceding section, which involves overwriting the cache.